posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+        private GreeterGrpc.GreeterBlockingStub greetingService;
+
+        private void startService(String address, int port) 
+        {
+            channel = ManagedChannelBuilder.forAddress(address , port).usePlaintext().build();
+            greetingService = GreeterGrpc.newBlockingStub(channel);
+        }
+
+        private void stopService() 
+        {
+            channel.shutdownNow();
+        }
+
+        @Override
+        protected void doGet(HttpServletRequest reqest, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+            // set user, address, port params
+        }
+
+        @Override
+        protected void doPost(HttpServletRequest request, HttpServletResponse response) 
+            throws ServletException, IOException 
+        {
+
+        // grab user, address, port params
+        startService(address, port);
+        HelloRequest person = HelloRequest.newBuilder().setName(user).build();
+        HelloReply greeting = greetingService.sayHello(person);
+
+        // send the greeting in a response
+        stopService();
+        }	
+    }
+}
+----
+
+As with the service example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. All required gRPC client libraries are provided by `grpcClient-1.0`.
+
+[#bugs]
+== Significant bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A20.0.0.12+label%3A%22release+bug%22[full list of bugs fixed in 20.0.0.12].
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14542[IllegalAccessError when using MP Rest Client with Java 15]
++
+There was a MP Rest Client failure when running with Java 15. This was fixed by changing the visibility of the `compareCustomStatus` method to public. If you would like to know more about MP Rest Client check out our link:https://openliberty.io/guides/microprofile-rest-client.html[Consuming RESTful services with template interfaces] and link:https://openliberty.io/guides/microprofile-rest-client-async.html[Consuming RESTful services asynchronously with template interfaces] guides.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14650[MP GraphQL does not scan JARs in WEB-INF/lib for GraphQL components]
++
+Classes annotated with things like `@GraphQLApi`, `@Type`, etc. or types referenced from root level queries that exist in JARs in the WEB-INF/lib directory were not processed. This meant that only classes in the WEB-INF/classes directory were processed by the MP GraphQL runtime. 
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14582[Prevent jsonp-1.0 and jsonpContainer-1.1 from both starting]
```

Comment:
If we're not going to link to the json-p guide, I'm not sure we should even both with this issue given how minor it is (it's not as much as functional product bug as it as a serviceability one).  Also, if we do want to keep this one, lets move to be the last defect listed.

Reply:
I've moved it to the end.